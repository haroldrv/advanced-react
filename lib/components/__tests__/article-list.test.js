import React from 'react';
import ArticleList from '../article-list';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('article list', () => {
  it('renders correctly', () => {
    const testProps = {
      articles: {
        a: { id: 'a' },
        b: { id: 'b' }
      },
      articleActions: {
        lookupAuthor: jest.fn(() => ({}))
      }
    };
    const renderer = new ShallowRenderer();
    renderer.render(<ArticleList {...testProps}></ArticleList>);
    const result = renderer.getRenderOutput();

    console.log(result);
    expect(result.props.children.length).toBe(2);
    //expect(tree).toMatchSnapshot();
  });
});