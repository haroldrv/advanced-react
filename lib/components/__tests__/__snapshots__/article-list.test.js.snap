// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`article list renders correctly 1`] = `
<div>
  <div />
  <div />
  <br />
  <div>
    <div>
      <a />
    </div>
  </div>
  <div />
  <div />
  <br />
  <div>
    <div>
      <a />
    </div>
  </div>
</div>
`;
